### PR TITLE
Fix Windows JVM crash when file-system watching is enabled

### LINKED ISSUE #13135: JVM crash with file-system watching enabled on Windows
`ArtifactTransformCachingIntegrationTest.cache cleanup does not delete entries that are currently being created` seems to crash in `file-events.dll`.

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  EXCEPTION_ACCESS_VIOLATION (0xc0000005) at pc=0x00007ff90d7b3b69, pid=10600, tid=8716
#
# JRE version: OpenJDK Runtime Environment (14.0+36) (build 14+36-1461)
# Java VM: OpenJDK 64-Bit Server VM (14+36-1461, mixed mode, sharing, tiered, compressed oops, g1 gc, windows-amd64)
# Problematic frame:
# C  [native-platform-file-events.dll+0x23b69]
```

See https://builds.gradle.org/viewLog.html?buildId=34884728&buildTypeId=Gradle_Check_VfsRetention_30_bucket6&fromExperimentalUI=true

---
cc: @gradle/build-cache

### LINKED ISSUE #13366: Cannot cancel watching on Windows due to "device or resource busy"
Sometimes cancelling a watch point fails on Windows with
```
Couldn't cancel watch point <work-dir>\subprojects\ide\build\tmp\test files\CompositeBuildIdeaProjectIntegrationTest\builds_IDEA_metadat...ame_build\8484g\buildA: device or resource busy: device or resource busy
```
See for example https://e.grdev.net/s/r7hh2az6r74qy/tests/:ide:vfsRetentionIntegTest/org.gradle.plugins.ide.idea.CompositeBuildIdeaProjectIntegrationTest/builds%20IDEA%20metadata%20with%20substituted%20dependency%20from%20same%20build#1

It seems that Gradle starts watching the project root and then very shortly after stops watching it again, most probably because nothing is left to watch in the project root.

See https://github.com/gradle/gradle/issues/13135#issuecomment-633445781.

---
cc: @gradle/execution

### PR DESCRIPTION
**Pull request by lptr**
Fixes #13135
Fixes #13366

This only aims to fix the crash, not the underlying `device or resource busy` issue when closing the watched directory handle.

### CODE REVIEW

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy [hunk 1]
```diff
@@ -16,19 +16,15 @@
 
 package org.gradle.integtests.resolve.verification
 
-
```
Comment (wolfs) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy:19]: Nit: extra new line.

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy [hunk 2]
```diff
@@ -16,19 +16,15 @@
...
-import org.gradle.integtests.fixtures.ToBeFixedForVfsRetention
 import org.gradle.integtests.fixtures.cache.CachingIntegrationFixture
 import org.gradle.test.fixtures.file.TestFile
-import org.gradle.util.TestPrecondition
 import spock.lang.Issue
 import spock.lang.Unroll
 
 import static org.gradle.util.Matchers.containsText
 
-@ToBeFixedForVfsRetention(because = "https://github.com/gradle/gradle/issues/13135", failsOnlyIf = TestPrecondition.WINDOWS)
```
Comment (wolfs) [inline subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy:31]: There are a few more ignored test on `master` now. There is also `DependenciesAttributesIntegrationTest` and `GroovyJavaLibraryInteractionIntegrationTest`. I wonder if `RichConsoleCompositeBuildGroupedTaskFunctionalTest` should work again as well.

File: subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy (hunk 1 above)
Comment (lptr) [reply subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/verification/DependencyVerificationIntegrityCheckIntegTest.groovy:19]: Hm?

File: gradle/dependencies.gradle [hunk 3]
```diff
@@ -77,7 +77,7 @@ libraries.kryo =                [coordinates: 'com.esotericsoftware.kryo:kryo',
 libraries.maven3 =              [coordinates: 'org.apache.maven:maven-core', version: '3.0.5', license: "Apache 2.0"]
 libraries.maven3_wagon_file =   [coordinates: 'org.apache.maven.wagon:wagon-file', version: '3.0.0', because: '3.1.0 of wagon-http seems to break Digest authentication', license: "Apache 2.0"]
 libraries.maven3_wagon_http =   [coordinates: 'org.apache.maven.wagon:wagon-http', version: libraries.maven3_wagon_file.version, license: "Apache 2.0"]
-libraries.nativePlatform =      [coordinates: 'net.rubygrapefruit:native-platform', version: '0.22-milestone-3', license: "Apache 2.0"]
+libraries.nativePlatform =      [coordinates: 'net.rubygrapefruit:native-platform', version: '0.22-snapshot-20200615134452+0000', license: "Apache 2.0"]
```
Comment (wolfs) [inline gradle/dependencies.gradle:80]:
```suggestion
libraries.nativePlatform =      [coordinates: 'net.rubygrapefruit:native-platform', version: '0.22-snapshot-20200615140910+0000', license: "Apache 2.0"]
```

Is this the version from your branch? Can you use `0.22-snapshot-20200615140910+0000` which is a snapshot from `master` where we archived the DLLs and PDB files?

Comment (lptr) [reply gradle/dependencies.gradle]: Done.

**Pull request merged**